src/access/profiling/payu_manager.py

```diff
@@ -27,6 +44,86 @@ def get_component_logs(self, path: Path) -> dict[str, ProfilingLog]:
             dict[str, ProfilingLog]: Dictionary mapping component names to their ProfilingLog instances.
         """
 
+    @property
+    def nruns(self) -> int:
+        """Returns the number of repetitions for the Payu experiments.
+
+        Returns:
+            int: Number of repetitions.
+        """
+        return self._nruns
+
+    @nruns.setter
+    def nruns(self, value: int) -> None:
+        """Sets the number of repetitions for the Payu experiments.
+
+        Args:
+            value (int): Number of repetitions.
+        """
+        if value < 1:
+            raise ValueError("Number of runs must be at least 1.")
+        self._nruns = value
+
+    @property
+    def startfrom_restart(self) -> str:
+        """Returns the restart option for the Payu experiments.
+
+        Returns:
+            str: Restart option.
+        """
+        return self._startfrom_restart
+
+    @startfrom_restart.setter
+    def startfrom_restart(self, value: str) -> None:
+        """Sets the restart option for the Payu experiments.
+
+        Args:
+            value (str): Restart option.
+        """
+        self._startfrom_restart = value
+
+    def generate_experiments(self, branches: list[str]) -> None:
+        """Generates Payu experiments for profiling data generation.
+
+        Args:
+            branches (list[str]): List of branches to generate experiments for.
+        """
+
+        for branch in branches:
+            if branch in self.experiments:
+                logger.info(f"Experiment for branch {branch} already exists. Skipping addition.")
+            else:
+                self.experiments[branch] = ProfilingExperiment(self.work_dir / branch / self.config_name)
+
+    def run_experiments(self) -> None:
+        """Runs Payu experiments for profiling data generation."""
+
+        runner_config = {
+            "test_path": self.work_dir,
+            "repository_directory": self.config_name,
```

Comment:
So `config_name` needs to be valid path, and not an arbitrary string

Separately, experiment-generator created the branches in the `repository_directory`, based on `model_type` (from the test-parsers jupyter notebook) - does that matter here?

Reply:
My repo directory was originally blank - so may be that's how experiment-generator needed the model-type. 

And yup, those are the directory layouts I get as well